### PR TITLE
cephfs: add support for encryption in ceph-csi-cephfs chart

### PR DESCRIPTION
**Pull request by NymanRobin**
The chart currently lacks access to configmap and secrets this causes the mounting of encrypted file systems to fail

To enable the chart to support the encrypted file systems add the following permissions:

Gives the cephfs nodeplugin clusterrole get access to configmaps and secrets.
While also adding cephfs provisioner get access to configs

Fixes issue: #4470 

This still needs the kernel version 6.6 to be able to be tested, I should be able to get this tested.

### CODE REVIEW

File: charts/ceph-csi-cephfs/templates/nodeplugin-clusterrole.yaml [hunk 1]
```diff
@@ -14,4 +14,10 @@ rules:
   - apiGroups: [""]
     resources: ["nodes"]
     verbs: ["get"]
+  - apiGroups: [""]
+    resources: ["configmaps"]
+    verbs: ["get"]
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "list"]
```
Comment (iPraveenParihar) [inline charts/ceph-csi-cephfs/templates/nodeplugin-clusterrole.yaml]: just the `get` action would be sufficient I believe.

Comment (NymanRobin) [reply charts/ceph-csi-cephfs/templates/nodeplugin-clusterrole.yaml]: Thanks for noticing, you are correct. I will fix it!

**Review by nixpanic — APPROVED** [review]

**Review by iPraveenParihar — APPROVED** [review]

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
@mergifyio queue

**Pull request merged**